gpu/gpu.go

```diff
@@ -65,31 +66,32 @@ func GetGPUInfo() GpuInfo {
 	}
 
 	var memInfo C.mem_info_t
-	resp := GpuInfo{"", "", 0, 0}
+	resp := GpuInfo{"", 0, 0}
 	if gpuHandles.cuda != nil {
 		C.cuda_check_vram(*gpuHandles.cuda, &memInfo)
 		if memInfo.err != nil {
 			log.Printf("error looking up CUDA GPU memory: %s", C.GoString(memInfo.err))
 			C.free(unsafe.Pointer(memInfo.err))
 		} else {
-			resp.Driver = "CUDA"
-			resp.Library = "cuda_server"
+			resp.Library = "cuda"
 		}
 	} else if gpuHandles.rocm != nil {
 		C.rocm_check_vram(*gpuHandles.rocm, &memInfo)
 		if memInfo.err != nil {
 			log.Printf("error looking up ROCm GPU memory: %s", C.GoString(memInfo.err))
 			C.free(unsafe.Pointer(memInfo.err))
 		} else {
-			resp.Driver = "ROCM"
-			resp.Library = "rocm_server"
+			resp.Library = "rocm"
 		}
 	}
-	if resp.Driver == "" {
+	if resp.Library == "" {
 		C.cpu_check_ram(&memInfo)
-		resp.Driver = "CPU"
 		// In the future we may offer multiple CPU variants to tune CPU features
-		resp.Library = "default"
+		if runtime.GOOS == "windows" {
+			resp.Library = "cpu"
+		} else {
+			resp.Library = "default"
```

Comment:
I'm actually changing this a little on another branch for windows native support which will layer on this PR.
The intent is "default" is the built-in binary, but a "cpu" library was dynamic.  In the case of windows, even the CPU model is dynamically loaded, but on linux/mac, default is built/linked directly in to the Go binary.

Reply:
Does this mean only supporting macOS 10.11 and later (i.e. macOS versions which support metal)? I think that's probably reasonable given metal has been out for >9 years.

Reply:
I don't believe my change has a material impact on our mac compat matrix in its current form, as the metal lib works on intel and falls back with the CPU.  That said, the pattern I've established would allow us to open up support for older hardware if we chose too, e.g., https://developer.nvidia.com/nvidia-cuda-toolkit-11_7_0-developer-tools-mac-hosts could be supported if there was user demand, but I believe the vram on those old systems is rather meager, so it's probably not worth it.